lib/AST/CMakeLists.txt

```diff
@@ -55,11 +55,10 @@ add_swift_library(swiftAST STATIC
   TypeRepr.cpp
   TypeWalker.cpp
   USRGeneration.cpp
-  LINK_LIBRARIES
+  LINK_LIBS
     swiftMarkup
     swiftBasic
 
-  INTERFACE_LINK_LIBRARIES
```

Comment:
`LINK_LIBRARIES` as implemented by Swift's `_add_swift_library_single` required a separate dentation for interface dependencies. LLVM's `llvm_add_library` does not because it does the right thing for `LINK_LIBS`. Any library dependencies specified on static archives become interface dependencies.

I would expect this patch to work correctly on Windows, for the same reason LLVM has never had the problem that you were working around.